src/workloads/scale/LargeIndexedInsMatchingDocuments.yml

```diff
@@ -0,0 +1,84 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/product-perf"
+Description: |
+  This is an indexed $in workload that matches documents in a collection.
+  This workload is intended to test a more representative workload
+
+  An extremelly common usecase for users is to fetch objects and hydrate with follow up $in query
+
+  For example:
+  The first query fetches an Author object
+  {
+    type: "author",
+    name: "Tyler",
+    posts: [ObjectId(1), ObjectId(2), ObjectId(3)],
+  }
+
+  The second query fetches the related, but not embedded/denormalized objects
+
+  { $in: [ObjectId(1), ObjectId(2), ObjectId(3)] }
+
+  This workload focuses on the second query of this use case matching documents
+  using arrays of integers that are generated to be used for the $in queries.
+Clients:
+  Default:
+    QueryOptions:
+      maxPoolSize: 20000
+  Update:
+    QueryOptions:
+      maxPoolSize: 100
+
+Actors:
+  - Name: BlogPostsLoader
+    Type: Loader
+    Threads: 1
+    Phases:
+      - Repeat: 1
+        Database: indexedins
+        CollectionCount: 1
+        Threads: 1
+        DocumentCount: 2048000
+        BatchSize: 100
+        Document:
+          _id: { ^Inc: { start: 0 } }
+          title: &randomString { ^RandomString: { length: 8 } }
+          text: *randomString
+      - { Nop: true }
+
+  - Name: FindBlogPosts
+    Type: CrudActor
+    Threads: 64
+    Database: indexedins
+    Phases:
+      - { Nop: true }
+      - Name: MatchDocumentsUsingIn
+        Duration: 2 minutes
+        Collection: Collection0
+        Operations:
+          - OperationName: find
+            OperationCommand:
+              Filter:
+                _id:
+                  {
+                    ^Array:
+                      {
+                        of:
+                          {
+                            ^Inc:
+                              {
+                                start: { ^RandomInt: { min: 0, max: 2028000 } }
+                              },
+                          },
+                        number: 20000,
```

Comment:
Wow, this is likely to be a very heavy weight test!

Have you tested it to see how noisy it is likely to be or if it generates stable results?